pkg/controller/operators/operatocondition_controller.go

```diff
@@ -0,0 +1,247 @@
+		},
+	}
+	err = r.Client.Create(context.TODO(), &roleBinding)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+			return err
+		}
+		existingRoleBinding := rbacv1.RoleBinding{}
+		err := r.Client.Get(context.TODO(), client.ObjectKey{Name: roleBinding.GetName(), Namespace: roleBinding.GetNamespace()}, &existingRoleBinding)
+		if err != nil {
+			return err
+		}
+		existingRoleBinding.OwnerReferences = roleBinding.OwnerReferences
+		existingRoleBinding.Subjects = roleBinding.Subjects
+		existingRoleBinding.RoleRef = roleBinding.RoleRef
+		err = r.Client.Update(context.TODO(), &existingRoleBinding)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func hasOwnerFunc(ownerReference metav1.OwnerReference) func(object metav1.Object) bool {
+	return func(object metav1.Object) bool {
+		if ownerutil.HasOwnerReference(object, ownerReference) {
+			return true
+		}
+		return false
+	}
+}
+
+func filterServiceAccounts(serviceAccounts []corev1.ServiceAccount, f func(object metav1.Object) bool) []corev1.ServiceAccount {
+	// // filter based on OwnerReferences
+	filteredServiceAccounts := []corev1.ServiceAccount{}
+	for _, serviceAccount := range serviceAccounts {
+		if f(&serviceAccount) {
+			filteredServiceAccounts = append(filteredServiceAccounts, serviceAccount)
+		}
+	}
+	return filteredServiceAccounts
+}
+
+func (r *OperatorConditionReconciler) ensureDeploymentEnvVars(operatorCondition *operatorsv1.OperatorCondition) error {
+	// filter based on OwnerReferences
+	for _, deploymentName := range operatorCondition.Spec.Deployments {
+		deployment := &appsv1.Deployment{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: deploymentName, Namespace: operatorCondition.GetNamespace()}, deployment)
+		if err != nil {
+			return err
+		}
+		for i := range deployment.Spec.Template.Spec.Containers {
+			deployment.Spec.Template.Spec.Containers[i].Env = ensureEnvVar(corev1.EnvVar{Name: OperatorConditionEnvVarKey, Value: operatorCondition.GetName()}, deployment.Spec.Template.Spec.Containers[i].Env)
+		}
+		err = r.Client.Update(context.TODO(), deployment)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func filterDeployments(deployments []appsv1.Deployment, f func(object metav1.Object) bool) []appsv1.Deployment {
+	// // filter based on OwnerReferences
+	filteredDeployments := []appsv1.Deployment{}
+	for _, deployment := range deployments {
+		if f(&deployment) {
+			filteredDeployments = append(filteredDeployments, deployment)
+		}
+	}
+	return filteredDeployments
+}
+
+func ensureEnvVar(envVar corev1.EnvVar, envVars []corev1.EnvVar) []corev1.EnvVar {
+	if len(envVars) == 0 {
+		return []corev1.EnvVar{envVar}
+	}
+
+	result := []corev1.EnvVar{}
+	found := false
+	for i := range envVars {
+		if envVars[i].Name != envVar.Name {
+			result = append(result, envVars[i])
+		} else {
+			found = true
+			result = append(result, envVar)
+		}
+	}
+
+	if !found {
+		result = append(result, envVar)
+	}
+
+	return result
```

Comment:
If I understand, this can be simplified a bit (and more if the duplication isn't important):

```suggestion
	dupe := make([]corev1.EnvVar, len(envVars))
	copy(dupe, envVars)

	for _, each := range dupe {
		if each.Name == envVar.Name {
			return dupe
		}
	}

	return append(dupe, envVar)
```

Reply:
Addressed in the latest version of the PR.